### PR TITLE
Use user id and project name as part of file paths for indexeddb storage

### PR DESCRIPTION
**Pull request by gideonthomas**

### CODE REVIEW

**Comment by gideonthomas** [comment]
This allows us to maintain different file paths for storage on bramble and in the publish.wm.org database. Thimble server acts as the moderator for this. Every incoming file path from publish-->Thimble server will be the file path as is. On Thimble server, the file path will be stored (in the session) based on whatever publish sends it. Every outgoing path from Thimble server-->Thimble client will be the stored path in Thimble server **prepended by `/:userID/projects/:projectID`** so that bramble can use that path to store all the project files

**Comment by gideonthomas** [comment]
This patch depends on https://github.com/mozilla/publish.webmaker.org/pull/115 landing

File: public/scripts/editor/js/fc/load-project-files.js [hunk 1]
```diff
@@ -22,56 +22,26 @@ define(["jquery", "constants", "text!fc/stay-calm/index.html", "text!fc/stay-cal
     var projectFilesUrl = "//" + window.location.host + "/initializeProject";
     var request;
     var projectTitle = project && project.title;
+    var root = project && project.root;
     var defaultProject = false;
-    var defaultPath;
     var files;
 
-    if(!projectTitle) {
+    if(!projectTitle || !root) {
```
Comment (sedge) [inline public/scripts/editor/js/fc/load-project-files.js]: Is it possible to have one but not the other?

File: public/scripts/editor/js/fc/load-project-files.js [hunk 2]
```diff
@@ -187,9 +157,11 @@ define(["jquery", "constants", "text!fc/stay-calm/index.html", "text!fc/stay-cal
     });
   }
 
-  function generateDefaultProject(title) {
+  function generateDefaultProject(title, root) {
+    title = title || Constants.ANON_PROJECT_NAME;
```
Comment (sedge) [inline public/scripts/editor/js/fc/load-project-files.js]: Nit: Why this change?

File: public/scripts/editor/js/fc/load-project-files.js [hunk 3]
```diff
@@ -187,9 +157,11 @@ define(["jquery", "constants", "text!fc/stay-calm/index.html", "text!fc/stay-cal
...
     return {
-      title: title || Constants.ANON_PROJECT_NAME,
+      root: root || Path.join("/", title),
```
Comment (sedge) [inline public/scripts/editor/js/fc/load-project-files.js]: It's doubly odd since you only do it for the one parameter.

File: public/scripts/editor/js/fc/load-project-files.js (hunk 1 above)
Comment (gideonthomas) [inline public/scripts/editor/js/fc/load-project-files.js]: realistically, this condition should never eval to true because that would be very bad, I wonder if there is even a point checking for this

File: public/scripts/editor/js/fc/load-project-files.js (hunk 3 above)
Comment (gideonthomas) [inline public/scripts/editor/js/fc/load-project-files.js]: I don't, next line uses it as well

File: public/scripts/editor/js/fc/load-project-files.js (hunk 2 above)
Comment (gideonthomas) [inline public/scripts/editor/js/fc/load-project-files.js]: I use it in two places below

**Comment by sedge** [comment]
r+
